### PR TITLE
console: migrate button to TypeScript

### PR DESCRIPTION
**Pull request by kawamataryo**
### Description

Related issue #4314.
Migrated button to TypeScript.

### Changelog

- [ ] `CHANGELOG.md` is updated with user-facing content relevant to this PR.

### Affected components

- [ ] Server
- [x] Console
- [ ] CLI
- [ ] Docs
- [ ] Community Content
- [ ] Build System
- [ ] Tests
- [ ] Other (list it)

### Related Issues
Related issue #4314.

#### Catalog upgrade

Does this PR change Hasura Catalog version?
- [x] No
- [ ] Yes

#### Metadata

Does this PR add a new Metadata feature?
- [x] No
- [ ] Yes

#### GraphQL
- [x] No new GraphQL schema is generated
- [ ] New GraphQL schema is being generated:

#### Breaking changes

- [x] No Breaking changes
- [ ] There are breaking changes:

### CODE REVIEW

**Comment by akhi5614** [comment]
How to use call bomb???

File: console/src/components/UIKit/atoms/Button/Button.ts [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+  BackgroundColorProps,
+} from 'styled-system';
+import { Theme } from '../../theme';
+
+interface OmittedButtonProps
+  extends Omit<
+    React.ComponentPropsWithRef<'button'>,
+    'type' | 'color' | 'disabled'
+  > {}
+const Button = ('button' as any) as React.FC<OmittedButtonProps>;
```
Comment (beerose) [inline console/src/components/UIKit/atoms/Button/Button.ts:24]:
```suggestion
const Button = ('button' as any) as React.FC<ButtonProps>;
```

File: console/src/components/UIKit/atoms/Button/Button.ts [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+  ${layout}
+  ${space}
+  ${typography}
+  ${color}
+  ${border}
+  ${flexbox}
+`;
+
+export interface StyledButtonProps
+  extends OmittedButtonProps,
```
Comment (beerose) [inline console/src/components/UIKit/atoms/Button/Button.ts:88]:
```suggestion
  extends ButtonProps,
```

File: console/src/components/UIKit/atoms/Button/Button.ts [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+    TextColorProps,
+    BackgroundColorProps,
+    TypographyProps,
+    SpaceProps,
+    FlexboxProps,
+    BorderProps {
+  boxShadowColor: string;
+  type: keyof Theme['button'];
+  disabled: boolean;
+  opacity: string | undefined;
```
Comment (beerose) [inline console/src/components/UIKit/atoms/Button/Button.ts:37]:
```suggestion
  opacity?: string;
```

**Comment by kawamataryo** [comment]
@beerose 

Thank you for your review.
I fixed as suggest.
https://github.com/hasura/graphql-engine/pull/4570/commits/5b4afeb8a152281d1eea8ce2335613f7a75644aa

File: console/src/components/UIKit/atoms/Button/Button.ts (hunk 3 above)
Comment (kawamataryo) [reply console/src/components/UIKit/atoms/Button/Button.ts]: Fix in 5b4afeb

File: console/src/components/UIKit/atoms/Button/Button.ts (hunk 2 above)
Comment (kawamataryo) [reply console/src/components/UIKit/atoms/Button/Button.ts]: Fix in 5b4afeb

File: console/src/components/UIKit/atoms/Button/Button.ts (hunk 1 above)
Comment (kawamataryo) [reply console/src/components/UIKit/atoms/Button/Button.ts]: Fix in 5b4afeb

File: console/src/components/UIKit/atoms/Button/Button.ts [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+  SpaceProps,
+  TypographyProps,
+  BorderProps,
+  FlexboxProps,
+  TextColorProps,
+  BackgroundColorProps,
+} from 'styled-system';
+import { Theme } from '../../theme';
+
+interface OmittedButtonProps
```
[the remark this answers is not in the document]
Comment (kawamataryo) [reply console/src/components/UIKit/atoms/Button/Button.ts]: Fix in 5b4afeb

**Pull request merged**